src/Server.php

```diff
@@ -162,6 +165,7 @@ public function handleRequest(ConnectionInterface $conn, RequestInterface $reque
             $stream = new LengthLimitedStream($stream, $contentLength);
         }
 
+        $request = $request->withoutHeader('Content-Length');
```

Comment:
What is the point in always removing this? Isn't this something (for example) a middleware might be interested in?

Also, I don't see this documented anywhere, in particular not the *WHY*?